projects/dendrites/subnetworks/utils.py

```diff
@@ -0,0 +1,383 @@
+# flake8: noqa
+
+import math
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+
+from sklearn import datasets
+from sklearn.preprocessing import MinMaxScaler
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import accuracy_score
+from torch import nn
+
+from nupic.research.frameworks.dendrites import DendriteSegments
+
+__all__ = [
+    "split_data",
+    "apply_target_transform",
+    "add_task_embedding",
+    "prep_dataset",
+    "load_datasets",
+    "evaluate",
+    "train",
+    "run_experiment",
+    "plot_activations",
+    "StandardMLP",
+    "DendriticMLP",
+]
+
+# ------ Dataset
+def split_data(X, Y, random_state=None):
+    """Split with random selection
+    Return two tuples, one with X and another with Ys"""
+    X_train, X_test, Y_train, Y_test = train_test_split(
+        X, Y, test_size=0.2, random_state=random_state, stratify=Y)
+    return (X_train, X_test), (Y_train, Y_test)
+
+def apply_target_transform(Y, offset=0):
+    """Transform label for continual learning problems"""
+    return [label + offset for label in Y]
+
+
+def add_task_embedding(X, task_num=1, units=2):
+    """Adds a task embedding to the input"""
```

Comment:
Can you add more detail in the docstring about what this function does?

Reply:
Done